### ISSUE TITLE
Elements shifted off the page with negative locations are considered visible

### ISSUE THREAD

**Issue by twalpole** [thoughtbot/capybara-webkit#951]
When using capybara-webkit (and poltergeist)

    <input id="radio_1" name="radio" value="1" type="radio", style="position:absolute; left: -9999px>

is considered a visible element, although it cannot be seen or interacted with (clicked, etc).  Current selenium-webdriver (2.53.4) reports it as non-visible, although the 3.0.0.beta3.1 version reports it as visible but throws an ElementNotVisible error if you try to click it.    Not really sure what the current outcome here is, although I lean towards that it should be non-visible.

**Comment by jferris** [comment]
Here's what we currently do for a click test: https://github.com/thoughtbot/capybara-webkit/blob/master/src/WebPage.cpp#L286

We could additionally check to see that the coordinates we're trying to click are on the screen (or can be scrolled to).

**Comment by mhoran** [comment]
We also check that the node is visible in this code: https://github.com/thoughtbot/capybara-webkit/blob/464da6915173bd5adcebee88362d1e84e0751396/src/capybara.js#L251-L260 which is called from [`Capybara.clickPosition`](https://github.com/thoughtbot/capybara-webkit/blob/464da6915173bd5adcebee88362d1e84e0751396/src/capybara.js#L203).

**Comment by twalpole** [comment]
@jferris Yes - it currently errors correctly if one tries to click the element - the issue (if there is an issue) is that if #visible? is called on the element it returns true.   I'm not sure whether or not that should be the case since the user can't usually scroll to negative positions to see the element.

**Comment by twalpole** [comment]
Note: this only came up because of the :allow_label_click option for checkboxes/radio buttons in Capybara 2.8+ which capybara-webkit doesn't yet support, and I'm working around it for a 2.9.1 release, but it would be nice to have a consensus on what the result of #visible? should be for these elements.

**Comment by jferris** [comment]
@mhoran if we used `clickTest` when asking if an element was visible, that would change the original example to return `false`, right?

**Comment by mhoran** [comment]
Yeah, I think so. Though something may be visible, but obscured by another element, in which case `clickTest` would return false. So I guess we should just add bounds checking to `isNodeVisible`.

**Comment by jferris** [comment]
Hmm, actually, that's interesting: if something is obscured behind a modal dialog or something, is it visible?

**Comment by twalpole** [comment]
I believe selenium considers  something visible if the user can scroll it into a position where it can be seen, so  behind a modal could still be considered visible, of course that may be changing as selenium moves towards only supporting browsers through vendor provided drivers, and the webdriver spec solidifies.

**Comment by mhoran** [comment]
@jferris I was also thinking something along the lines of an overlaying semi-transparent (or completely transparent div), that allows the underling content to be visible, but not clickable.
